include/circt/Dialect/HW/HWAttributes.td

```diff
@@ -119,6 +119,23 @@ def FileListAttr : AttrDef<HWDialect, "FileList"> {
 
 }
 
+/// An attribute used to attach a comment to an operation.  This exists,
+/// independently of things like `sv::VerbatimOP` to anchor a comment and
+/// prevent it becoming detached from its operation.
+def CommentAttr : AttrDef<HWDialect, "Comment"> {
```

Comment:
Unlike AttributeAttr, this is literally just a StringAttr.  I don't think we need an additional CommentAttr wrapper.  Do you see a benefit of having this as opposed to just using StringAttr directly?

Reply:
With the current implementation, no. There's no benefit. There are future extensions that would benefit from this being a separate attribute, e.g., a formatting option bit indicating _one line comment_/_block comment_ or support for symbol substitution.

Does this justify a separate attribute or should we hill climb to this from `StringAttr` if needed?

Reply:
in IR design, it is always a good idea to start with the minimal solution to a problem and add complexity later.  There is frequently pressure to think ahead beyond the current hill to what you think is over the horizon on the next hill and overdesign for it.  However, often you never get to the next hill, or when you do, you find out that you've learned a lot more and need a different solution.

As an example from a different discussion, I see the same concern with verilog parsing.  When we do a verilog parser, the SV dialect will be insufficient, so we'll need to do another thing.  I don't think that "this might be useful for a verilog parser" to be sufficient motivation to add complexity to the SV dialect.  When we get there, we will understand the goals and constraints better.

Reply:
This specific case is a bit different - it is "just" adding code complexity for no specific benefit.  The cost is lower than proactively adding IR complexity, but I still don't think it is warranted.   Going with a simple solution means less code to maintain and evolve etc.

Reply:
Makes sense to me. 👍  Thanks for laying out your thought process on IR design here.

Reply:
CommentAttr is now gone in favor of a `StringAttr` on `HWModuleOp` in the rebase.